tests/acceptance/focus-test.js

```diff
@@ -103,3 +103,36 @@ test('Should load leaf routes with a spurious focusing outlet without any errors
   visit('/about');
   andThen(() => { assert.equal(checkFocus(), 'About Us'); });
 });
+
+test('Transition abort and redirect behavior.', function(assert) {
+  // transition.abort();
+  visit('/');
+  visit('/messages');
+  visit('/transition-root/abort');
+  andThen(() => { assert.equal(checkFocus(), 'Messages'); });
+
+  // this.transitionTo('feed');
+  visit('/');
+  visit('/transition-root/redirect/feed');
+  andThen(() => { assert.equal(checkFocus(), 'Feed'); });
```

Comment:
This is the test that fails. Somehow not calling `transition.abort()` explicitly makes this behave differently.